src/front/components/vehicles/cardList.vue

```diff
@@ -0,0 +1,31 @@
+<script setup lang="ts">
+import type { Vehicle } from '~/models/Vehicle';
+
+const enum Availability {
+    All,
+    Available,
+    Unavailable
+}
+const props = defineProps({
+    type: Number,
+});
+
+const type = ref(props.type)
+
+const shouldDisplayVehicle = (vehicle: Vehicle): Boolean => {
```

Comment:
C'est une bonne pratique de n'utiliser que les types primitifs, sauf exception bien précise. 
https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#the-primitives-string-number-and-boolean